aws/resource_aws_ec2_traffic_mirror_target.go

```diff
@@ -0,0 +1,120 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsEc2TrafficMirrorTarget() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEc2TrafficMirrorTargetCreate,
+		Read:   resourceAwsEc2TrafficMirrorTargetRead,
+		Delete: resourceAwsEc2TrafficMirrorTargetDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"network_interface_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ConflictsWith: []string{
+					"network_load_balancer_arn",
+				},
+			},
+			"network_load_balancer_arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ConflictsWith: []string{
+					"network_interface_id",
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsEc2TrafficMirrorTargetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	input := &ec2.CreateTrafficMirrorTargetInput{}
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("network_interface_id"); ok {
+		input.NetworkInterfaceId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("network_load_balancer_arn"); ok {
+		input.NetworkLoadBalancerArn = aws.String(v.(string))
+	}
+
+	out, err := conn.CreateTrafficMirrorTarget(input)
+	if err != nil {
+		return fmt.Errorf("Error creating traffic mirror target %v", err)
+	}
+
+	d.SetId(*out.TrafficMirrorTarget.TrafficMirrorTargetId)
+
+	return resourceAwsEc2TrafficMirrorTargetRead(d, meta)
+}
+
+func resourceAwsEc2TrafficMirrorTargetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	targetId := d.Id()
+	input := &ec2.DescribeTrafficMirrorTargetsInput{
+		TrafficMirrorTargetIds: []*string{&targetId},
+	}
+
+	out, err := conn.DescribeTrafficMirrorTargets(input)
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("Error describing traffic mirror target %v", targetId)
+	}
+
+	if 1 != len(out.TrafficMirrorTargets) {
+		d.SetId("")
+		return fmt.Errorf("Not able to find find target %v", targetId)
+	}
```

Comment:
Same note here -- when triggering resource recreation we should prefer a warning log and return no error.

```go
if out == nil || len(out.TrafficMirrorTargets) == 0 {
	log.Printf("[WARN] EC2 Traffic Mirror Target (%s) not found, removing from state", d.Id())
	d.SetId("")
	return nil
}
```